### ISSUE TITLE
Support for security logging annotations and aspects

### ISSUE THREAD

**Issue by javabeanz** [javabeanz/owasp-security-logging#5]

**Comment by spoofzu** [comment]
Hi @javabeanz.  Please provide a little more detail for the benefits of logging annotations and aspects so we understand what you propose.  So for example, what annotations do you suggest and what do they do?  Also for AOP, what aspects should we create and where should they be bound?  Thanks.  --Milton

**Comment by javabeanz** [comment]
all configuration for the securrity logging library now is done now with xml, but it can also be done with annotations. Here is an example of what i mean : https://projectlombok.org/features/log. If you can annotate a property with say @Masked it would only show up in logs as masked value (f.i. last 4 numbers of a credit card). Those are the features i am thinking of.

**Comment by augustd** [comment]
I like this idea. Developers could annotate properties of their classes and prevent them from being logged incidentally (e.g. if `MyBean.toString()` is called).

**Comment by spoofzu** [comment]
I agree, Log and Masked appear beneficial features.  It’s tiring to constantly define the same class variable everywhere.  I like minimizing broiler-plate code.  —Milton

Please excuse if terse.  Sent via mobile device.
